### DISCUSSION TITLE
升级到v4 后还能使用子模块吗

### DISCUSSION THREAD

**Discussion by Halecoder** [oldwinter/knowledge-garden#84] [提问] [answered]
如题，我想子模块更新内容后，自动发布到3版本和4版本的contents文件加中，还有，此ci.yaml文件中的变量需要自己在GitHub下设置吗

**Comment by oldwinter** [comment, marked as answer]
https://quartz.jzhao.xyz/hosting  可以看一下作者的部署文档哦，部署到vercel上面，用户访问速度会比较快。我没有用github action，所以不确定还是否需要ci.yaml那个文件。
v4还可以继续用子模块的方案的，现在有个麻烦点就是，我想实时同步作者的最新修改，但是一旦作者的最新修改改动了content里面的内容，就会导致我们要解决冲突。关于这点，我也想在抽空的时候再好好设计一下，将来如何能更好地更快地实时同步作者的最新版本。

**Reply by Halecoder** [reply]
我才发现，作者content文件夹好像没有内容，都是在 docs里，不知有啥区别，还有请问，您没有用 github action,那您是咋用vercel 部署的，cloud flare吗

**Reply by Halecoder** [reply]
解决了，vercel和cf部署都不需要进行github action，只需要在.gitmodule中把子模块链接改成https://{token}@github.com/{username}/{repo}.git 即可，要不然无法拉去取子模块，作者提供了构建命令，npx quartz build只要记得加上以及改下子模块文件中的GitHub链接即可，不像v3先用工作流输出git pages ，再进行vercel 托管

**Comment by chelseachen007** [comment]
> 解决了，vercel和cf部署都不需要进行github action，只需要在.gitmodule中把子模块链接改成https://{token}@github.com/{username}/{repo}.git 即可，要不然无法拉去取子模块，作者提供了构建命令，npx quartz build只要记得加上以及改下子模块文件中的GitHub链接即可，不像v3先用工作流输出git pages ，再进行vercel 托管

你好请问下这个 token 怎么加上去，我看你和 oldwinter 的 gitmodules 里都没有加 也不适合加
